### ISSUE TITLE
Code error on startup

### ISSUE THREAD

**Issue by Dark007mario** [OpenNBS/OpenNoteBlockStudio#348]
Anytime I go to open Open

**To Reproduce**
1. I open the program
2. It crashes and gives an error code

**Expected behavior**
The program to run normally.

**Screenshots**
![Error Code](https://user-images.githubusercontent.com/49248328/184511855-cbcba167-376f-4c61-baed-92754ea63ea7.PNG)

**Additional context**
Here is the code it spits out:

___________________________________________
############################################################################################
ERROR in
action number 1
of Draw Event
for object obj_controller:

index out of bounds request 11 maximum size is 11############################################################################################
gml_Script_midi_input_set_key (line 8)
gml_Script_draw_piano (line 155)
gml_Script_control_draw (line 2228)
gml_Object_obj_controller_Draw_0 (line 0)

**Comment by SoftBlade** [comment]
do you did something that made the program crash? like modifying the files?

**Comment by MericGit** [comment]
Do you happen to have any midi devices plugged in? Both "virtual" and actual connected devices.

At the bottom of your screen NBS should list the MIDI devices it detects
![image](https://user-images.githubusercontent.com/41242144/184517810-1f8d30c9-3c68-45c1-970e-f7805249a499.png)

Can you take a look there and let me know what appears?

**Comment by Dark007mario** [comment]
> do you did something that made the program crash? like modifying the files?

No this was a fresh install

**Comment by Dark007mario** [comment]
> Do you happen to have any midi devices plugged in? Both "virtual" and actual connected devices.
> 
> At the bottom of your screen NBS should list the MIDI devices it detects ![image](https://user-images.githubusercontent.com/41242144/184517810-1f8d30c9-3c68-45c1-970e-f7805249a499.png)
> 
> Can you take a look there and let me know what appears?
> 
> Looks similar to issue #237 at first glance

I cant even get to this, this is the first thing that pops up on launch

**Comment by MericGit** [comment]
> > Do you happen to have any midi devices plugged in? Both "virtual" and actual connected devices.
> > At the bottom of your screen NBS should list the MIDI devices it detects ![image](https://user-images.githubusercontent.com/41242144/184517810-1f8d30c9-3c68-45c1-970e-f7805249a499.png)
> > Can you take a look there and let me know what appears?
> > Looks similar to issue #237 at first glance
> 
> I cant even get to this, this is the first thing that pops up on launch

It still shows up in the bottom bar.

**Comment by Dark007mario** [comment]
> > > Do you happen to have any midi devices plugged in? Both "virtual" and actual connected devices.
> > > At the bottom of your screen NBS should list the MIDI devices it detects ![image](https://user-images.githubusercontent.com/41242144/184517810-1f8d30c9-3c68-45c1-970e-f7805249a499.png)
> > > Can you take a look there and let me know what appears?
> > > Looks similar to issue #237 at first glance
> > 
> > 
> > I cant even get to this, this is the first thing that pops up on launch
> 
> Do you know if you've ever installed any other music software or used midi devices? (I.E. OrbComposer)

Yes I do a lot with midi. I am a regular Ableton and Reaper user for controlling live shows. I use a variety of controllers (mostly Launchpads, apc40, midi fighter). I use Bomes midi translator quite a lot along side midi loop back and rtpmidi for show control. I also use a Mif4 Timecode interface to convert and stream LTC into MTC for timecode control in a few diffrent apps.

**Comment by MericGit** [comment]
> > > > Do you happen to have any midi devices plugged in? Both "virtual" and actual connected devices.
> > > > At the bottom of your screen NBS should list the MIDI devices it detects ![image](https://user-images.githubusercontent.com/41242144/184517810-1f8d30c9-3c68-45c1-970e-f7805249a499.png)
> > > > Can you take a look there and let me know what appears?
> > > > Looks similar to issue #237 at first glance
> > > 
> > > 
> > > I cant even get to this, this is the first thing that pops up on launch
> > 
> > 
> > Do you know if you've ever installed any other music software or used midi devices? (I.E. OrbComposer)
> 
> Yes I do a lot with midi. I am a regular Ableton and Reaper user for controlling live shows. I use a variety of controllers (mostly Launchpads, apc40, midi fighter). I use Bomes midi translator quite a lot along side midi loop back and rtpmidi for show control. I also use a Mif4 Timecode interface to convert and stream LTC into MTC for timecode control in a few diffrent apps.

Yeah it's highly likely then that some other musical application created a bunch of virtual midi devices which nbs cannot handle. Pinning down exactly which one it is will be difficult if the bottom bar in nbs doesn't even show up, but one of the nbs devs might be able to assist you in figuring it out
 
Unfortunately I myself aren't too experienced on that end so you'll probably need to wait for one of them, sorry

**Comment by Dark007mario** [comment]
> > > > > Do you happen to have any midi devices plugged in? Both "virtual" and actual connected devices.
> > > > > At the bottom of your screen NBS should list the MIDI devices it detects ![image](https://user-images.githubusercontent.com/41242144/184517810-1f8d30c9-3c68-45c1-970e-f7805249a499.png)
> > > > > Can you take a look there and let me know what appears?
> > > > > Looks similar to issue #237 at first glance
> > > > 
> > > > 
> > > > I cant even get to this, this is the first thing that pops up on launch
> > > 
> > > 
> > > Do you know if you've ever installed any other music software or used midi devices? (I.E. OrbComposer)
> > 
> > 
> > Yes I do a lot with midi. I am a regular Ableton and Reaper user for controlling live shows. I use a variety of controllers (mostly Launchpads, apc40, midi fighter). I use Bomes midi translator quite a lot along side midi loop back and rtpmidi for show control. I also use a Mif4 Timecode interface to convert and stream LTC into MTC for timecode control in a few diffrent apps.
> 
> Yeah it's highly likely then that some other musical application created a bunch of virtual midi devices which nbs cannot handle. Pinning down exactly which one it is will be difficult if the bottom bar in nbs doesn't even show up, but one of the nbs devs might be able to assist you in figuring it out
> 
> Unfortunately I myself aren't too experienced on that end so you'll probably need to wait for one of them, sorry

Yeah I figured since it only gives an error code and it looks more like a bug in code as supposed to something I did with using a midi device. I am not a programmer but my blind guess is something wrong with either drawing/rendering something in the gui. Thanks for your attempt

**Comment by MericGit** [comment]
> > > > > > Do you happen to have any midi devices plugged in? Both "virtual" and actual connected devices.
> > > > > > At the bottom of your screen NBS should list the MIDI devices it detects ![image](https://user-images.githubusercontent.com/41242144/184517810-1f8d30c9-3c68-45c1-970e-f7805249a499.png)
> > > > > > Can you take a look there and let me know what appears?
> > > > > > Looks similar to issue #237 at first glance
> > > > > 
> > > > > 
> > > > > I cant even get to this, this is the first thing that pops up on launch
> > > > 
> > > > 
> > > > Do you know if you've ever installed any other music software or used midi devices? (I.E. OrbComposer)
> > > 
> > > 
> > > Yes I do a lot with midi. I am a regular Ableton and Reaper user for controlling live shows. I use a variety of controllers (mostly Launchpads, apc40, midi fighter). I use Bomes midi translator quite a lot along side midi loop back and rtpmidi for show control. I also use a Mif4 Timecode interface to convert and stream LTC into MTC for timecode control in a few diffrent apps.
> > 
> > 
> > Yeah it's highly likely then that some other musical application created a bunch of virtual midi devices which nbs cannot handle. Pinning down exactly which one it is will be difficult if the bottom bar in nbs doesn't even show up, but one of the nbs devs might be able to assist you in figuring it out
> > Unfortunately I myself aren't too experienced on that end so you'll probably need to wait for one of them, sorry
> 
> Yeah I figured since it only gives an error code and it looks more like a bug in code as supposed to something I did with using a midi device. I am not a programmer but my blind guess is something wrong with either drawing/rendering something in the gui. Thanks for your attempt

You can actually disregard the UI portion of the error log, it's just due to the way GML (The language the program is written in) works. I'm like 99% certain the issue is with MIDI devices, chances are you have too many virtual midi devices that NBS isn't smart enough to ignore, so it loads em all and can't handle them.  You might be able to use a third party program to find all midi devices and maybe from there determine how to resolve it?

**Comment by Bentroen** [comment]
I'm leaving this one open since I've seen a few people report this issue in the past. Even if NBS can't deal with some of the connected devices, it should "fail safely" and display a nice error message, but not crash.
